### PR TITLE
Montana Child Tax Credit

### LINKED ISSUE #2326: Montana Child Tax Credit
https://leg.mt.gov/bills/2023/billhtml/HB0268.htm

### PR DESCRIPTION
**Pull request by Kevintowne**
### Summary
:sparkles::notebook::test_tube:

This pull request adds a new feature to the policy engine: the Montana child tax credit. It defines the parameters, variables, and tests for the credit, and updates the changelog and the income-sources notebook accordingly.

> _`Montana` credit_
> _New feature for tax season_
> _YAML and Python_

### Walkthrough
*  Add Montana child tax credit feature ([link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-6b7db4d36edc25bc780f127368867649a7360ecec8d90a23bacd90b90b03b36fL1-L-1), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-d4c7065350cfb904f560da9ae130a5a1760314128aad8114865946133d4e29ffR1-R25), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-9e57ff4f13bc4ab85a4683882be022e8b34657adf3c29d7ba045215ec3488e0fR1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-954266dd34656ba8ca5e0631f00334906458f401756791ecc702d4e64cd6b0deR1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-07e4f24e51c8fa32d5e9d666e973c4764ac7d34aacd7ca5866f228b4cee8ed80R1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-9f1ab252c6e710cd9d77fdc228f0995ad815a159f70668c3f2611d608e994a49R1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-837342844a4ef9e3a68d542f257339d5f4d77dc3f69aafb1dfa6fc40f4095d2eR1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-eaf3abd8da41cc1c9932529282bab8147bd7ee18985465b137d0bb68fb96d763R1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-ae2a752a17be53826a7c01c3187ceb097f9898196e8f6cab4d569981e398e8daR1-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-4d150a80e0ce559fa6266b7bda12ad31349549f16a8d1170d07c266e4d0a2f96R1-R24), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-cd3f5a76d48307cb755f64df087c35950dcb3115d3242e17b7ac310dd365e519R1-R30), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-c5ca9c26182588ca4b95f72b4429077d095f7e9b3e836c8d826c8584aa5439c5R1-R11))
* Reformat code in `income-sources.ipynb` to improve readability and follow black code style ([link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-39f32c3f894f7b5b41bdccc557d6fb450370fa89876e96f01f8d2903cd0ffb77L2057-R2068), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-39f32c3f894f7b5b41bdccc557d6fb450370fa89876e96f01f8d2903cd0ffb77L2085-R2099), [link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-39f32c3f894f7b5b41bdccc557d6fb450370fa89876e96f01f8d2903cd0ffb77L2093-R2117))
* Fix typo in `income-sources.ipynb` ([link](https://github.com/PolicyEngine/policyengine-us/pull/3105/files?diff=unified&w=0#diff-39f32c3f894f7b5b41bdccc557d6fb450370fa89876e96f01f8d2903cd0ffb77L2121-R2138))

Fixes #2326

### CODE REVIEW

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:3]:
```suggestion
  2023-01-01:
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml:3]:
```suggestion
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml:3]: need a threshold?

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml:3]: make a refundable.yaml and include this credit for 2023

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml:12]:
```suggestion
      2023-01-01: 0
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml:6]:
```suggestion
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml:4]:
```suggestion
      2023-01-01: 0
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:14]: exact part of the law

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:3]:
```suggestion
  2023-01-01:
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:15]: add part of Montana law that points to the disqualified income

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/threshold.yaml:12]: exact section of the law

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/threshold.yaml:12]: check all yaml files

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:24]: delete

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/threshold.yaml
Comment (Kevintowne) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/threshold.yaml]: Thank you for the comments. I'm a bit confused about what do you mean by exact section of the law? Do you mean legal codes related to this threshold other than 2023 Montana Legislature, House Bill No. 268, Section 1, 6? Thank you!

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:1]: could we move this to the prameters/irs/gross_income folder

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:16]: delete montana source

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income/disqualified_income.yaml:4]: maybe its just divident income?

File: policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py:4]:
```suggestion
class disqualified_income(Variable):
```

File: policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py:7]:
```suggestion
    label = "Disqualified income"
```

File: policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py:8]: add reference to IRS file

File: policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/mt_disqualified_income.py:1]: same as parameter - move to IRS

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/investment.yaml:14]:
```suggestion
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:18]:
```suggestion
        # CTC limited to filers with investment income below a certain threshold
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml:1]: remove the eligibility folder and just keep the income_limit

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/agi.yaml:7]:
```suggestion
  label: Montana child tax credit adjusted gross income limit
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/eligibility/income_limit/earned_income_required.yaml:8]:
```suggestion
  label: Montana child tax credit earned income limit
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/reduction/amount.yaml:7]:
```suggestion
  label: Montana child tax credit reduction amount
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:24]:
```suggestion
        earned_income_eligible = earned_income > p.earned_income_required
```

**Comment by MaxGhenis** [comment]
please request my review after @PavelMakarchuk approves

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/agi.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/agi.yaml:11]:
```suggestion
    - title: 2023 Montana Legislature, House Bill No. 268, Section 1. 7 (d)
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/agi.yaml:11]: please double check all sections so that they align with the parameter values

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/agi.yaml:11]: approved after that

File: policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml:1]: Not Montana, this is federal. Add a label

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py:22]:
```suggestion
        reduction = p.reduction.amount * increments
```
Rate implies percent

File: policyengine_us/variables/gov/irs/gross_income/disqualified_income.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]: Disqualified from what?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/irs/gross_income/disqualified_income.py
Comment (Kevintowne) [reply policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]: image
The legal code mentioned the definition of investment income as the disqualified income mentioned in the internal revenue code. So I think they represent the investment income for Montana state.

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]: earned income disqualified income ([section 32 IRS](https://www.law.cornell.edu/uscode/text/26/32))
@Kevintowne please adjust the parameter description and label to represent that

image

Comment (Kevintowne) [reply policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]: Got it. I'll fix it.

File: policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml:1]:
```suggestion
description: Income sources counted as disqualified income for the earned income calculation.
```

File: policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml:13]:
```suggestion
  label: Earned income disqualified income sources
```

File: policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml:12]:
```suggestion
  unit: list
  period: year
```

File: policyengine_us/variables/gov/irs/gross_income/disqualified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]:
```suggestion
class earned_income_disqualified_income(Variable):
```

File: policyengine_us/variables/gov/irs/gross_income/disqualified_income.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:1]: rename the file to `earned_income_disqualified_income`

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:1]: rename file in tests and make sure everything passes

File: policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/gross_income/disqualified_income.yaml:3]:
```suggestion
  2021-01-01:
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/amount.yaml:1]:
```suggestion
description: Montana provides the following base amount per child under the child tax credit, based on age.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/earned_income_required.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/earned_income_required.yaml:1]: dont need this parameter but have to check for earned_income > 0 in the variable

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py:18]:
```suggestion
        # Credit gets reduced by an amount for each increment that AGI exceeds a certain threshold 
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:25]:
```suggestion
        earned_income_eligible = earned_income > 0
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: please request my review once all is done

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:25]:
```suggestion
        receives_earned_income = earned_income > 0
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc_eligible.py:17]:
```suggestion
        agi_eligible = agi <= p.agi
```

File: policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/credits/ctc/mt_ctc.py:17]:
```suggestion
        credit_amount = tax_unit.sum(p.amount.calc(age))
```

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/README.md
Comment (ThapeloMasasa) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/README.md:1]: Please delete this file.. it is not needed here

File: policyengine_us/variables/gov/irs/gross_income/disqualified_income.py
Comment (ThapeloMasasa) [inline policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]: is it possible to avoid using "income" twice in the same variable name.. more like "disqualified_earned_income"

Comment (ThapeloMasasa) [inline policyengine_us/variables/gov/irs/gross_income/disqualified_income.py:4]: Also please change the file name to match the class name for all variable files

File: policyengine_us/variables/gov/irs/gross_income/disqualified_earned_income.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/gross_income/disqualified_earned_income.py:1]: Don't need this or the parameter, it's in eitc_relevant_investment_income

https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/irs/credits/earned_income/eitc_relevant_investment_income.py

File: policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/agi.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/credits/ctc/income_limit/agi.yaml:1]:
```suggestion
description: Montana limits its child tax credit to filers with federal adjusted gross income below this amount.
```

**Review by MaxGhenis — APPROVED** [review]

**Comment by MaxGhenis** [comment]
good to merge after these

**Pull request merged**
